python/huggingface_server_cpu_openvino.Dockerfile

```diff
@@ -0,0 +1,77 @@
+ARG BASE_IMAGE=ubuntu:22.04
+ARG VENV_PATH=/prod_venv
+
+FROM ${BASE_IMAGE} AS builder
+
+
+# Install Poetry
+ARG POETRY_HOME=/opt/poetry
+ARG POETRY_VERSION=1.8.3
+
+# Install vllm
+ARG VLLM_VERSION=v0.6.4.post1
+
+RUN apt-get update -y && apt-get install -y \
+    gcc python3.10-venv python3-dev python3-pip \
+    ffmpeg libsm6 libxext6 libgl1 libnuma1 git \
+    && apt-get clean && rm -rf /var/lib/apt/lists/*
+
+# Set up Poetry for dependency management
+RUN python3 -m venv ${POETRY_HOME} && ${POETRY_HOME}/bin/pip3 install poetry==${POETRY_VERSION}
+ENV PATH="$PATH:${POETRY_HOME}/bin"
+
+# Activate virtual env
+ARG VENV_PATH
+ENV VIRTUAL_ENV=${VENV_PATH}
+RUN python3 -m venv $VIRTUAL_ENV
+ENV PATH="$VIRTUAL_ENV/bin:$PATH"
+
+# Install KServe and Hugging Face Server dependencies
+COPY kserve/pyproject.toml kserve/poetry.lock kserve/
+RUN cd kserve && poetry install --no-root --no-interaction --no-cache
+COPY kserve kserve
+RUN cd kserve && poetry install --no-interaction --no-cache
+
+COPY huggingfaceserver/pyproject.toml huggingfaceserver/poetry.lock huggingfaceserver/health_check.py huggingfaceserver/
+RUN cd huggingfaceserver && poetry install --no-root --no-interaction 
+COPY huggingfaceserver huggingfaceserver
+RUN cd huggingfaceserver && poetry install --no-interaction --no-cache
+
+# Clone vllm
+# Install Python build tools and other dependencies then build vllm from source
+WORKDIR /vllm
+RUN git clone --branch $VLLM_VERSION --depth 1 https://github.com/vllm-project/vllm.git . && \
+    pip install --upgrade pip && \
+    pip install -r requirements-build.txt --extra-index-url https://download.pytorch.org/whl/cpu && \
+    PIP_EXTRA_INDEX_URL="https://download.pytorch.org/whl/cpu" VLLM_TARGET_DEVICE="openvino" python -m pip install -v .
+
+FROM ubuntu:22.04 AS prod
+
+RUN apt-get update && apt-get upgrade -y && apt-get install python3.10-venv libnuma1 build-essential gcc python3-dev -y && apt-get clean && \
```

Comment:
Do we need gcc ?

Reply:
kept gcc in prod since it is present in prod for huggingface_server image also: https://github.com/kserve/kserve/blob/c9cc812c9ecb07da915f12c85f36d8b83d57ccd0/python/huggingface_server.Dockerfile#L39

Reply:
I think we don't need them there as well. @yuzisun

Reply:
Actually we need these packages in prod since vLLM requires them, they were added in this PR: https://github.com/kserve/kserve/pull/3802